bazel/emscripten_deps.bzl

```diff
@@ -94,11 +97,65 @@ filegroup(
         ],
     ),
 )
+
+write_file(
+    name = "embuilder_config",
+    out = "emscripten_config",
+)
+
+genrule(
+    name = "embuilder",
+    tools = [
+        ":emscripten/embuilder.py",
+        ":compiler_files",
+        ":ar_files"
+    ],
+    srcs = [":embuilder_config"],
+    cmd = \"\"\"
+export EM_BINARYEN_ROOT=$$(realpath $$(dirname $$(dirname $(location :emscripten/embuilder.py))))
+export EM_LLVM_ROOT=$$EM_BINARYEN_ROOT/bin
+export EM_EMSCRIPTEN_ROOT=$$EM_BINARYEN_ROOT/emscripten
+export EM_CACHE=$(RULEDIR)/emscripten/cache
```

Comment:
These values normally part of the config file.   See https://github.com/emscripten-core/emsdk/blob/main/bazel/emscripten_toolchain/emscripten_config

I think the only one you want to override here is EM_CACHE.   The others should all come from that config file.

Reply:
The problem there (I think/I will test this) is that `emscripten_config` is part of the `@emsdk//` while this code is inside `@emscripten_bin_XXX//` so this might create a circular dependency.

> I think the only one you want to override here is EM_CACHE. The others should all come from that config file.

I would also need to set `EM_FROZEN_CACHE=0` since at this point the cache should be writable.